xmppserver/src/main/java/org/jivesoftware/openfire/net/DNSUtil.java

```diff
@@ -90,28 +91,39 @@ public class DNSUtil {
      *
      * If that lookup fails as well, it's assumed that the XMPP server lives at
      * the host resolved by a DNS A lookup at the specified domain on the
-     * specified default port.<p>
+     * specified default port.
      *
      * As an example, a lookup for "example.com" may return "im.example.com:5269".
      *
+     * The returned collection is a list of sets of host names. The 'inner'
+     * collection, the sets of host names, are grouping host names that have an
+     * equal 'priority' SRV value. The 'outer' collection, the list, defines the
+     * order that's defined by the 'priority' value (lowest first). Generally
+     * speaking, all values from the first set of the outer list should be
+     * processed, before elements from the next set of that list are to be processed.
+     *
+     * The Sets that are returned as elements of the list have a predictable
+     * iteration order. This order is based on a randomization based on the
+     * SRV 'weight' value of each host.
+     *
      * @param domain the domain.
      * @param defaultPort default port to return if the DNS look up fails.
      * @return a list of  HostAddresses, which encompasses the hostname and port
```

Comment:
Trivial, but this bit of Javadoc is no longer correct.

Reply:
How is this not correct? The list returns Sets that are instances of LinkedHashSet, which retain insertion order - or at least, that's the idea.

The purpose of this is to have a collection that, once randomized at initialization, has a predictable iteration order when it's being used.

I was debating if it would be more appropriate to use a List or Set here. List would be more indicative of a strict iteration order, while Set usage hints at the uniqueness of each element that it contains.

Reply:
JavaDoc says you are returning a list of HostAddresses. 
In practice, you're returning a list of Sets. Each Set contains a set of HostAddresses.
Iterating over the response requires different behaviour.

Reply:
Oooh, I was looking at what GitHub renders as the green bit :) Sure, lets fix that.